### PR TITLE
TP template cleanups and fixes for future SW

### PR DESCRIPTION
**Pull request by jasonfan393**
Removes some unnecessary templates on main TP function, moving them to be calculated with Seed/iTC template parameters. 

Fixes mismatch between NCOPY template parameter on VMSTE memories between VMRCM and TP. Moves some variable declarations outside main TP function loop (needed for future SW).

### CODE REVIEW

**Comment by jasonfan393** [comment]
CI checks appear to be in unrelated integration test chains

File: TestBenches/TrackletProcessor_test.cpp
Comment (aryd) [inline TestBenches/TrackletProcessor_test.cpp:99]: We should see if this can be done in a cleaner way (but should not hold up this PR - can yo enter as an issue?)

File: TrackletAlgorithm/TrackletProcessor.h
Comment (aryd) [inline TrackletAlgorithm/TrackletProcessor.h:1015]: Fix formatting?

**Review by aryd — COMMENTED** [review]
Just some minor comment about formatting and a request to create an issue. After these are addressed we should merge.

File: TestBenches/TrackletProcessor_test.cpp
Comment (jasonfan393) [reply TestBenches/TrackletProcessor_test.cpp]: I've just replaced all the NTEUnits with NCOPY instead, since NTEUNITs is no longer a template parameter of TP there was no need to use it to determine ncopy in the testbench.

**Review by aryd — APPROVED** [review]
I'm happy with the code now.

**Comment by aryd** [comment]
This looks fine now - I will merge.

**Pull request merged**
